### PR TITLE
API: Add functions to retrieve version information

### PR DESCRIPTION
**Pull request by luebbers**
Adds API calls to retrieve version and build information from the
library.

### CODE REVIEW

File: libopae/src/version.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+fpga_result __FPGA_API__ fpgaGetVersionString(char *version_str, size_t len)
+{
+	if (!version_str) {
+		FPGA_MSG("build_str is NULL");
+		return FPGA_INVALID_PARAM;
+	}
+
+	strncpy(version_str, INTEL_FPGA_API_VERSION, len);
```
Comment (tswhison) [inline libopae/src/version.c]: Klocwork will want strncpy_s.

File: libopae/src/version.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+fpga_result __FPGA_API__ fpgaGetBuildString(char *build_str, size_t len)
+{
+	if (!build_str) {
+		FPGA_MSG("build_str is NULL");
+		return FPGA_INVALID_PARAM;
+	}
+
+	strncpy(build_str, INTEL_FPGA_API_HASH, len);
```
Comment (tswhison) [inline libopae/src/version.c]: Klocwork will want strncpy_s.

File: libopae/src/version.c (hunk 1 above)
Comment (luebbers) [reply libopae/src/version.c]: Right - will fix that. Thank you.

**Comment by luebbers** [comment]
Yes, the substitution was done in PR #93.

File: libopae/src/version.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+#include <config.h>
+#endif // HAVE_CONFIG_H
+
+#include "safe_string/safe_string.h"
+#include "common_int.h"
+#include "types_int.h"
+#include "config_int.h"
+
+#include <string.h>
+
```
Comment (r-rojo) [inline libopae/src/version.c]: Would it make sense to use the `fpga_version` struct we have defined in types? We could embed it here as a static variable initialized using the three macros...

```C
/**
 * Semantic version
 *
 * Data structure for expressing version identifiers following the semantic
 * versioning scheme. Used in various properties for tracking component
 * versions.
 */
typedef struct {
	uint8_t major;        /**< Major version */
	uint8_t minor;        /**< Minor version */
	uint16_t patch;       /**< Revision or patchlevel */
} fpga_version;
```

**Pull request merged**